orchestrator/coingecko/coingecko.go

```diff
@@ -49,6 +49,55 @@ func urlJoin(baseURL string, segments ...string) string {
 
 }
 
+func (cp *PriceFeed) QueryETHUSDPrice() (float64, error) {
+	u, err := url.ParseRequestURI(urlJoin(cp.config.BaseURL, "simple", "price"))
+	if err != nil {
+		cp.logger.Fatal().Err(err).Msg("failed to parse URL")
+	}
+
+	q := make(url.Values)
+
+	q.Set("ids", "ethereum")
+	q.Set("vs_currencies", "usd")
+	u.RawQuery = q.Encode()
+
+	reqURL := u.String()
+	req, err := http.NewRequest(http.MethodGet, reqURL, nil)
+	if err != nil {
+		cp.logger.Fatal().Err(err).Msg("failed to create HTTP request")
+	}
+
+	resp, err := cp.client.Do(req)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to fetch price from %s", reqURL)
+		return zeroPrice, err
+	}
+
+	respBody, err := ioutil.ReadAll(io.LimitReader(resp.Body, maxRespBytes))
+	defer resp.Body.Close()
+	if err != nil {
+		err = errors.Wrapf(err, "failed to read response body from %s", reqURL)
+		return zeroPrice, err
+	}
+
+	var f interface{}
```

Comment:
Is there a struct we can define and decode into instead? Remember, you don't need to define the entire JSON struct -- only the fields you're interested in.

Reply:
```js
// Response for ERC20 price
{"0xdac17f958d2ee523a2206206994597c13d831ec7":{"usd":0.998233}}

// Response for Ethereum price
{"ethereum": {"usd": 4271.57}}
```

I think we can do it for Ethereum but not for the tokens as the key is dynamic :(

Reply:
nvm, worked it out.

``` go
type priceResponse map[string]struct {
	USD float64 `json:"usd"`
}
```